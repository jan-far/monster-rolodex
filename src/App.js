import React, { Component } from "react";
import "./App.css";
import CardList from "./Components/Card-list";
import SearchBox from "./Components/SearchBox";

class App extends Component {
  constructor() {
    super();

    this.state = {
      monsters: [],
      searchField: "",
    };
  }

  componentDidMount() {
    fetch("https://jsonplaceholder.typicode.com/users")
      .then((response) => response.json())
      .then((users) => this.setState({ monsters: users }));
  }

  handleChange = (e) => {
    this.setState({ searchField: e.target.value });
  };

  filteredMonsters = () => this.state.monsters.filter((monster) =>
    monster.name.toLowerCase().includes(this.state.searchField.toLowerCase())
  );

  render() {
    return (
      <div className="App">
        <h1>Monsters Rolodex</h1>
        <SearchBox placeholder="Search Monters" handleChange={this.handleChange} />
        <CardList monsters={this.filteredMonsters} />
      </div>
    );
  }
}

export default App;
