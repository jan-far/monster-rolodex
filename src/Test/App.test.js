import React from "react";
import { shallow } from "enzyme";
import App from "../App";

const getMonters = async () => {
  const monters = await fetch("https://jsonplaceholder.typicode.com/users");
  const result = await monters.json();
  return result;
};

describe("App Tests", () => {
  const wrapper = shallow(<App />);

  it("Make a snapshot of App comonent", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should show initial value of the component state", () => {
    expect(wrapper.state()).toEqual({ monsters: [], searchField: "" });
  });

  it("should populate the monsters state of the app", () => {
    expect.assertions(1);

    return getMonters().then((data) => {
      wrapper.setState({ monsters: data });
      expect(wrapper.state("monsters").length).toEqual(data.length);
    });
  });

  it("should get undefined for handleChange function", () => {
    const event = { target: { value: "test" } };
    expect(wrapper.instance().handleChange(event)).toEqual(undefined);
  });

  it("should update the state of searchFied", () => {
    expect(wrapper.state("searchField")).toEqual("test");
  });

  it("should change state value for searchField", () => {
    const handleChange = jest.fn((val) => wrapper.setState({ searchField: val }));

    handleChange("monster");
    expect(wrapper.state("searchField")).toEqual("monster");
  });

  it("should filter the monsters state array", () => {
    const mockMonsters = [
      {
        id: 1,
        name: "ralph",
        email: "ralph@mail.com",
      },
      { id: 2, name: "flow", email: "flow@mail.com" },
    ];
    wrapper.setState({ monsters: mockMonsters });

    const handleChange = jest.fn((val) => wrapper.setState({ searchField: val }));
    handleChange("r");

    const filteredMonster = [
      {
        id: 1,
        name: "ralph",
        email: "ralph@mail.com",
      },
    ];

    expect(wrapper.instance().filteredMonsters()).toEqual(filteredMonster);
  });
});
