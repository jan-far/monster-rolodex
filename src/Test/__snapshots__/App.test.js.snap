// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests Make a snapshot of App comonent 1`] = `ShallowWrapper {}`;
